### ISSUE TITLE
start_metadata_sync_to_node() doesn't sync `hasmetadata` flag on existing worker's

### ISSUE THREAD

**Issue by onderkalaci** [citusdata/citus#2320]
Seems not critical, opening mostly for reference.

`start_metadata_sync_to_node()` should set the `hasmetadata` flag on the existing worker nodes. Note that `hasmetadata` field is currently only used on the coordinator node, so not syncing to the data nodes is not a problem as of 7.5.

Steps to reproduce:

```SQL
-- connect to the coordinator, add two worker node
SELECT master_add_node('localhost', 9700);
SELECT master_add_node('localhost', 9701);

-- start metadata sync
SELECT start_metadata_sync_to_node('localhost', 9700);
SELECT start_metadata_sync_to_node('localhost', 9701);

-- get the metadata table on the coordinator
SELECT nodename, nodeport, hasmetadata FROM pg_dist_node;

 nodename  | nodeport | hasmetadata 
-----------+----------+-------------
 localhost |     9700 | t
 localhost |     9701 | t
(2 rows)

-- connect to the first worker and get it
\c - - - 9700
SELECT nodename, nodeport, hasmetadata FROM pg_dist_node;
 nodename  | nodeport | hasmetadata 
-----------+----------+-------------
 localhost |     9701 | f
 localhost |     9700 | t
(2 rows)


-- connect to the second worker and get it
\c - - - 9701
SELECT nodename, nodeport, hasmetadata FROM pg_dist_node;
 nodename  | nodeport | hasmetadata 
-----------+----------+-------------
 localhost |     9700 | t
 localhost |     9701 | t
(2 rows)
```

As seen above, the metadata is out of sync for this field on the node with port `9700`.

**Comment by marcocitus** [comment]
I believe this was a conscious design decision to not add unnecessary complexity and exposure to failures of unrelated nodes in `start_metadata_sync_to_node`. 

One fundamental question in the design MX was: If you have HA+MX, how do you recover from multiple node failures, knowing that the only path to recovery is eventually updating the metadata on all nodes to point to the new primaries? Any operation that requires all nodes to be up will not be usable in such a scenario.

**Comment by onderkalaci** [comment]
Just a note that this could be tricky if we want to use `hasmetadata ` flag from the worker nodes. To give an example, we've first implemented #2345 but soon realized that due to this restriction (?), we should be implementing it like #2370 .

**Comment by marcocitus** [comment]
We will need to address this to enable features such as:
- DDL on distributed tables from workers
- CREATE TABLE on any node creates a Citus local table that is globally visible
